### PR TITLE
feat(TCOMP-2410) support document type

### PR DESCRIPTION
**Pull request by yyin-talend**
https://jira.talendforge.org/browse/TCOMP-2410

### CODE REVIEW

**Review by undx — COMMENTED** [review]
👍🏻 but a few comments.
We may need to handle document type in dynamic, this is not covered.
Check also Sonar errors and warnings, fix those possibles and mark the other as resolved.

File: component-studio/component-runtime-di/src/test/java/org/talend/sdk/component/runtime/di/record/DiRowStructVisitorTest.java [hunk 1]
```diff
@@ -225,6 +233,12 @@ void visit() {
         assertNull(record.getString("lookKey"));
     }
 
+    private void initDocument(Document document) {
+        org.dom4j.Document doc = DocumentHelper.createDocument();
+        doc.addElement("catelog").addComment("an XML catelog");
```
[the remark this answers is not in the document]
Comment (yyin-talend) [reply component-studio/component-runtime-di/src/test/java/org/talend/sdk/component/runtime/di/record/DiRowStructVisitorTest.java:241]: 2 places call this method.

File: component-studio/component-runtime-di/pom.xml [hunk 2]
```diff
@@ -59,7 +59,11 @@
       <scope>test</scope>
       <!-- rarely needed + don't deliver it by default since it brings a lot of CVE -->
     </dependency>
-
+    <dependency>
+      <groupId>org.dom4j</groupId>
+      <artifactId>dom4j</artifactId>
+      <version>${dom4j.version}</version>
```
[the remark this answers is not in the document]
Comment (yyin-talend) [reply component-studio/component-runtime-di/pom.xml:65]: Tested, somewhere still need dom4j, can't change the scope to test.

File: component-studio/component-runtime-di/src/test/java/org/talend/sdk/component/runtime/di/record/DiRowStructVisitorTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (undx) [reply component-studio/component-runtime-di/src/test/java/org/talend/sdk/component/runtime/di/record/DiRowStructVisitorTest.java]: was talking about `an XML catelog`...

**Review by undx — DISMISSED** [review]
see last comment remaining, otherwise lgtm.

File: component-studio/component-runtime-di/src/test/java/org/talend/sdk/component/runtime/di/record/DiRowStructVisitorTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (yyin-talend) [reply component-studio/component-runtime-di/src/test/java/org/talend/sdk/component/runtime/di/record/DiRowStructVisitorTest.java]: modified

**Review by undx — DISMISSED** [review]

**Review by undx — APPROVED** [review]

**Pull request merged**